civismlext/preprocessing.py

```diff
@@ -174,6 +174,15 @@ def _expand_col(self, X, col):
         # then it's a category not seen during the fit and should
         # be ignored in the expansion.
         catcol = X[col].astype('category')
+
+        # check for overlap between categories in col and original fit levels
+        overlap = len([lvl for lvl in catcol.cat.categories if
+                       lvl in self.levels_[col]])
```

Comment:
This would have better computational complexity if it were written as a set intersection (e.g. `overlap = len(catcol.cat.categories & set(self.levels_[col]))`. In the case where there's only a few categories on each side, I don't know what ends up being faster. I suspect the set intersection is probably faster, but I'm okay if you want to leave it as is.

Reply:
I timed out a few different options before going with this one (set(a).intersection(b), set & set, and this one), and the list comprehension timed as slightly faster than the other two, probably because it doesn't require casting to set. I only tried with a few categories, so it might be less efficient with many. I don't think it's going to be a major computational drain compared to the expansion itself, so I don't have a strong opinion. Happy to change it if you like the set notation better

Reply:
No, this works! Sorry for presuming you hadn't already tested this.